// berserk/webpack.config.js
const path = require('path')
const UglifyJSPlugin = require('uglifyjs-webpack-plugin')
const serverConfig = {
	entry:{
		jNode:'./index.js'
	},
	target: 'node',
	devtool: 'inline-source-map',
	output: {
		path: path.resolve(__dirname, 'dist'),
		filename: 'j.node.js',
		libraryTarget: 'umd'
	},
	devServer: {contentBase: './src'},
	plugins: [
		new UglifyJSPlugin()
	],
	module: {
		rules: [
			{
				test: /\.js$/,
				exclude: /(node_modules|bower_components)/,
				use: {
					loader: 'babel-loader',
					query: {
						presets:[ 'es2015','stage-2' ]
					}
				}
			}
		]
	}
}

const clientConfig = {
	entry:{
		jWeb:'./index.js'
	},
	target: 'web',
	devtool: 'inline-source-map',
	devServer: {contentBase: './src'},
	output: {
		path: path.resolve(__dirname, 'dist'),
		filename: 'j.web.js',
		libraryTarget: 'umd'
	},
	plugins: [
		new UglifyJSPlugin()
	],
	module: {
		rules: [
			{
				test: /\.js$/,
				exclude: /(node_modules|bower_components)/,
				use: {
					loader: 'babel-loader',
					query: {
						presets:[ 'es2015','stage-2' ]
					}
				}
			}
		]
	}

}

module.exports = [serverConfig, clientConfig]
